libfreerdp/core/nego.c

```diff
@@ -250,7 +250,8 @@ BOOL nego_security_connect(rdpNego* nego)
 		else if (nego->SelectedProtocol == PROTOCOL_SSL)
 		{
 			WLog_DBG(TAG, "nego_security_connect with PROTOCOL_SSL");
-			nego->SecurityConnected = transport_connect_tls(nego->transport);
+			nego->SecurityConnected =
+			    nego->transport->context->update->io->TLSConnect(nego->transport);
```

Comment:
I don´t like the exposure of the internal structure of `rdpTransport` you´re doing here.
We are working on moving all module/components to be self contained with only an opaque structure exposed, this moves in the opposite direction.

Why not simply move all these changes inside the old functions you just removed?